SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,430 @@
+bool FSetDefaultLayer::Update()
+{
+	ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(TestData->TestWorld->GetWorldSettings());
+	WorldSettings->DefaultLayerLoadBalanceStrategy = DefaultLayer;
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FAddLayer, TSharedPtr<TestData>, TestData, StrategyConfig, Strategy);
+bool FAddLayer::Update()
+{
+	ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(TestData->TestWorld->GetWorldSettings());
+	auto StratName = FName{ *FString::FromInt((WorldSettings->WorkerLayers.Num())) };
+	FLayerInfo LayerInfo;
+	LayerInfo.Name = StratName;
+	LayerInfo.LoadBalanceStrategy = Strategy.StrategyClass;
+	for (const auto& TargetActors : Strategy.TargetActorTypes) {
+		LayerInfo.ActorClasses.Add(TargetActors);
+	}
+	WorldSettings->WorkerLayers.Add(StratName, LayerInfo);
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSetupStrategy, TSharedPtr<TestData>, TestData, TOptional<uint32>, NumVirtualWorkers);
+bool FSetupStrategy::Update()
+{
+	ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(TestData->TestWorld->GetWorldSettings());
+	WorldSettings->DefaultLayerLoadBalanceStrategy = UGridBasedLBStrategy::StaticClass();
+	WorldSettings->bEnableMultiWorker = true;
+
+	auto& Strat = TestData->Strat;
+	Strat->Init();
+
+	if (!NumVirtualWorkers.IsSet()) {
+		NumVirtualWorkers = Strat->GetMinimumRequiredWorkers();
+	}
+
+	Strat->SetVirtualWorkerIds(1, NumVirtualWorkers.GetValue());
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_TWO_PARAMETER(FSetupStrategyLocalWorker, TSharedPtr<TestData>, TestData, VirtualWorkerId, worker_id);
+bool FSetupStrategyLocalWorker::Update()
+{
+	TestData->Strat->SetLocalVirtualWorkerId(worker_id);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FCheckWhoShouldHaveAuthority, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, FName, ActorName, VirtualWorkerId, Expected);
+bool FCheckWhoShouldHaveAuthority::Update()
+{
+	const VirtualWorkerId Actual = TestData->Strat->WhoShouldHaveAuthority(*TestData->TestActors[ActorName]);
+	Test->TestEqual(
+		FString::Printf(TEXT("Who Should Have Authority. Actual: %d, Expected: %d"), Actual, Expected),
+		Actual, Expected);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckMinimumWorkers, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, uint32, Expected);
+bool FCheckMinimumWorkers::Update()
+{
+	const uint32 Actual = TestData->Strat->GetMinimumRequiredWorkers();
+	Test->TestEqual(
+		FString::Printf(TEXT("Strategy for minimum required workers. Actual: %d, Expected: %d"), Actual, Expected),
+		Actual, Expected);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckStratIsReady, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, bool, Expected);
+bool FCheckStratIsReady::Update()
+{
+	const auto& Strat = TestData->Strat;
```

Comment:
In terms of using auto, we defer to Unreal's style guide which says yes in iterators and function/lambda types, otherwise no.

Reply:
removed